README.md

```diff
@@ -1,58 +1,2501 @@
-`ibmcloud ks cluster config -c <cluster>`
+2. Check that the Ingress controller is disabled. This might take a few minutes.
+```
+ibmcloud ks ingress alb get -c <cluster> --alb <alb-id>
+```
 
-* Modify the image reference in the deploy.yaml
- - In sample_deploy/deploy.yaml, change <image_reference> with the location of your image, i.e. <region>.icr.io/<namespace>/ingress:latest
+3. Get the local `kubeconfig` file with a valid token for your cluster.
+```
+ibmcloud ks cluster config -c <cluster>
+```
 
-* Copy image pull secret to the kube-system namespace
- - Since the service account for the ingress controller sample references the default pull secret in kube-system, it must be copied to the kube-system namespace
-`kubectl get secret all-icr-io -o yaml | sed 's/namespace: .*/namespace: kube-system/' | kubectl apply -f -`
- - note: If the all-icr-io secret does not exist in the default namespace, follow the instructions in this document to apply it: https://cloud.ibm.com/docs/containers?topic=containers-registry#imagePullSecret_migrate_api_key
+4. Modify the image reference in the `deploy.yaml` file. For example, in `sample_deploy/deploy.yaml`, replace `<image_reference>` with the location of your image, such as `<region>.icr.io/<namespace>/ingress:latest`.
 
-* Apply the deployment
-`kubectl apply -f ./sample_deploy`
+5. Copy the image pull secret to the `kube-system` namespace in your cluster. Because the service account for the Ingress controller sample references the default pull secret in `kube-system`, the secret must exist in the `kube-system` namespace.
+```
+kubectl get secret all-icr-io -o yaml | sed 's/namespace: .*/namespace: kube-system/' | kubectl apply -f -
+```
+> Note: If the `all-icr-io` secret does not exist in the default namespace, follow the instructions in this document to apply it: https://cloud.ibm.com/docs/containers?topic=containers-registry#imagePullSecret_migrate_api_key
 
-* Ensure the pods are Running
-`kubectl get po -n kube-system | grep ibm-cloud-ingress`
+6. Apply the deployment.
+```
+kubectl apply -f ./sample_deploy
+```
+
+7. Ensure the Ingress controller pods are in a `Running` state.
+```
+kubectl get po -n kube-system | grep ibm-cloud-ingress
+```
  - If not, check the events for errors:
-   `kubectl describe deploy -n kube-system ibm-cloud-ingress`
-
-### Register DNS domain
-* Get your service IP(Classic) or LB Domain(VPC)
-`kubectl get svc -n kube-system | grep ibm-cloud-ingress`
-  - note: The IP or domain is the 4th column
-
-* Create a DNS domain for your ingress
-  - for classic:
-`ibmcloud ks nlb-dns create classic -c <cluster> --ip <service ip address>`
-  - for VPC:
-`ibmcloud ks nlb-dns create vpc-gen2 -c <cluster> --lb-host <lb hostname>`
\ No newline at end of file
+   ```
+   kubectl describe deploy -n kube-system ibm-cloud-ingress
+   ```
+
+TODO: is this always going to be a public ingress controller w a public ip address (or hostname)? or can they specify private somehow?
```

Comment:
they could do private too if they wish.. While I provide only a public example, if somebody decides to use this and wants to on the private network, they should be able to easily modify the example to do this.